app/monitoring/Tip.scala

```diff
@@ -22,3 +24,38 @@ object TipFromConfig {
   }
 
 }
+
+object PathVerification {
+
+  type VerifyPath = String => Future[TipResponse]
+
+  sealed trait MonitoredRegion
+  case object AU extends MonitoredRegion
+  case object GB extends MonitoredRegion
+  case object US extends MonitoredRegion
+
+  sealed trait MonitoredProduct
+  case object OneOffContribution extends MonitoredProduct { override def toString: String = "One-off contribution" }
```

Comment:
overriding toString is my least favourite thing

Reply:
if you do really want to do it, and want to make it less boilerplatey you may be able to use a sealed abstract class instead of trait, and make it a parameter called toString.

Reply:
@johnduffell - I can make all of these vals (tipString?), but it's unclear to me how/why that's any better than overriding toString in this case. 

Would you mind explaining why overriding is a bad idea in this case? Happy to discuss this in person if it's easier.

Reply:
explained in person, basically toString exists on everything and it's not clear from the name what the purpose of it us (other than turn into some kind of string for unknown purpose) so would lose type safety of knowing that it's (probably) the right object.  `tipString` is a nice idea, or `value` - although the more we use `value`, the more that argument against it would apply there too!
Also I suggested using a sealed abstract class instead of trait, then the string can be a constructor param rather than an override with all the associated boiler plate.

Reply:
Thanks for summarising our discussion @johnduffell - I had a look at using abstract classes but decided I preferred sealed traits for this case.